proposal.rst

```diff
@@ -168,27 +210,83 @@ Project Details
 
 **Describe any existing work and concepts on which your project is based.**
 
-FILLME
+My work will be based on the GSoC 2017 Project `Clickstream Mining for 
+Learning to Rank training data <https://trac.xapian.org/wiki/GSoC2017/LetorClickstream>`_. 
+
+**Phase 1: Implementing Click Models**
+
+The previous work has implemented the simplified DBN model for clickstream mining. 
+I plan to finish the EM algorithm in order to get a better performance.
+
+
+Besides DBN model, I will implement two more click models, the Dependent click Model 
+(DCM) and the User Browsing Model (UBM). Both two are also highly 
+effective to improve document ranking. 
+
+I will first work on an abstract class for all click models and then try to implement 
+each model according to relevant papers. 
+
+
+
+**Phase 2: Documentation**
+
+The second phase of my work will be a detailed documentation of each model I 
+implemented. It will provide the Omega users with a brief view of click models and
+pros & cons of each model. I will continue to work on the clickmodel.rst in the 
+Omega documentation files.
+
+**Phase 3: Evaluation**
+
+The third phase is about the evaluation. With more and more models applied, I 
+believe it will be necessary to provide an evaluation method for the users. I'm 
+going to use CTR estimation for the evaluation of the models. For the method 
+is convenient to implement and has a relatively high accuracy. Also the click logs 
+is easy to acquire due to the previous GSoC project.
+
+
+
 
 **Do you have any preliminary findings or results which suggest that your
 approach is possible and likely to succeed?**
 
-FILLME
+The EM algorithm of **DBN model**: I will implement the EM algorithm for DBN model according to the appendix of 
+`A dynamic bayesian network click model for web search ranking. WWW (2009). 
+<https://dl.acm.org/citation.cfm?id=1526711>`_. 
+
+
+The **DCM model**: I will implement the dependent click model according to the paper `Efficient 
+multiple-click models in web search. WSDM (2009). 
+<https://dl.acm.org/citation.cfm?id=1498818>`_.
+
+The **UBM model**: the user browsing model is from paper `A user browsing model to predict search engine click data from past observations. SIGIR (2008).  <https://dl.acm.org/citation.cfm?id=1390392>`_.
+
 
 **What other approaches to have your considered, and why did you reject those in
 favour of your chosen approach?**
 
-FILLME
+The Federated Click Model (FCM) and Vertical Click Model (VCM)are not appropiate 
+for Xapian because of the SERP (search engine return pages) of Xapian do not 
+provide with vertical contents.
```

Comment:
I had to look up a paper on click models to figure out what this means. (I assumed "vertical" was to do with layout, rather than to do with category or type clustering, which was confusing.) A little bit more discussion of this would make things clearer. (It also opens the question of whether being able to run the query multiple times in OmegaScript — which would enable searches within verticals, but I believe is not currently possible — would be a good idea for the future.)

Reply:
There's collapsing which I guess can be used for a simple form of this, but certainly running two searches in one invocation is not currently possible in OmegaScript (one could probably implement this in JavaScript by combining two sets of search results fetched from separate Omega invocations via XMLHttpRequest for example).

However, it's not that **Xapian** SERPs don't provide vertical content, only that **Omega** ones don't.  It's entirely possible to implement a search system using Xapian that does, and people could generate click logs from such a system to feed into this.

I think it's probably better to focus on getting things working well for the common simple case though, especially as we couldn't easily test such models with Omega currently.

Reply:
Sorry for my misunderstanding. I will change my words and find a clear way to talk about it.